src/analysis/retail/deathknight/unholy/modules/Abilities.tsx

```diff
@@ -82,198 +65,230 @@ class Abilities extends CoreAbilities {
       },
       {
         spell: SPELLS.DEATH_AND_DECAY.id,
+        enabled: !combatant.hasTalent(TALENTS.DEFILE_TALENT),
         category: SPELL_CATEGORY.ROTATIONAL_AOE,
-        enabled: !combatant.hasTalent(talents.DEFILE_TALENT),
         cooldown: 30,
+        charges: combatant.hasTalent(TALENTS.DEATHS_ECHO_TALENT) ? 2 : 1,
         gcd: {
           base: 1500,
         },
       },
 
       // cooldowns
+
       {
-        spell: SPELLS.APOCALYPSE.id,
+        spell: TALENTS.APOCALYPSE_TALENT.id,
+        enabled: combatant.hasTalent(TALENTS.APOCALYPSE_TALENT),
         category: SPELL_CATEGORY.COOLDOWNS,
-        cooldown: combatant.hasTalent(talents.ARMY_OF_THE_DAMNED_TALENT) ? 45 : 90,
+        cooldown: 45,
         gcd: {
           base: 1500,
         },
-        castEfficiency: {
-          suggestion: true,
-          recommendedEfficiency: 0.9,
-          extraSuggestion: (
-            <span>
-              Making sure to use <SpellLink spell={SPELLS.APOCALYPSE} /> immediately after it's
-              cooldown is up is important, try to plan for it's use as it is coming off cooldown.
-            </span>
-          ),
-        },
+        // castEfficiency: {
+        //   suggestion: true,
+        //   recommendedEfficiency: 0.9,
+        //   extraSuggestion: (
+        //     <span>
+        //       Making sure to use <SpellLink spell={TALENTS.APOCALYPSE_TALENT} /> immediately after it's
+        //       cooldown is up is important, try to plan for it's use as it is coming off cooldown.
+        //     </span>
+        //   ),
+        // },
```

Comment:
generally should delete instead of just commenting out, unless you're planning to more or less immediately follow up on the change.